Public api spec cleanup

vsavkin opened the pull request:
- **Please check if the PR fulfills these requirements**
- [X] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit-message-format
- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

It simplifies the old public api spec to check only top-level symbols. This is to make sure that the Dart and TS apis do not diverge. The new public api spec verifies the TS api of Angular.
- **What is the current behavior?** (You can also link to an open issue here)

The old spec checks members.
- **What is the new behavior (if this is a feature change)?**

Does not check members.
- **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

No.
- **Other information**:

IgorMinar commented:
I'm fine with this. if @mhevery is ok with this change as well then go ahead and merge

mhevery commented:
LGTM

vsavkin committed:
cleanup(testing): clean up public api spec

Change the old public api spec to check only the exported top-level symbols. This will make sure that Dart and JS do not diverge. The new public api spec verifies the TS api.

mary-poppins commented:
Merging PR #7447 on behalf of @alexeagle to branch [presubmit-alexeagle-pr-7447](https://github.com/angular/angular/tree/presubmit-alexeagle-pr-7447).
